store/src/java/com/zimbra/cs/service/UserServlet.java

```diff
@@ -382,8 +382,7 @@ public void doGet(HttpServletRequest req, HttpServletResponse resp) throws Servl
             checkTargetAccountStatus(context);
 
             if (!checkTwoFactorAuthentication(context)) {
-                resp.addHeader(AuthUtil.WWW_AUTHENTICATE_HEADER, getRealmHeader(req, null));
-                resp.sendError(HttpServletResponse.SC_UNAUTHORIZED, L10nUtil.getMessage(MsgKey.errMustAuthenticate, req));
+                throw ServiceException.AUTH_REQUIRED(String.format("two factor auth is required"));
```

Comment:
I think you can pass a plain string instead of `String.format()` since there is nothing to replace right? or am I missing something?

Reply:
Agreed. Done.